tasks/typedoc.ts

```diff
@@ -27,34 +28,49 @@ function typedocOptions(options: any) {
 
 export = function (grunt: IGrunt) {
 	grunt.registerTask('typedoc', function (this: ITask) {
-		const deploy = process.env.DEPLOY_DOCS;
-		const shouldPublish = deploy === 'publish' || deploy === 'commit';
-
 		// Throw when any shelljs command fails
 		config.fatal = true;
 
 		const options: any = this.options({});
-		const rootApiDocDirectory = grunt.config.get<string>('apiDocDirectory');
-		const outOption = grunt.option<string>('doc-dir');
-		options.out = outOption || options.out || rootApiDocDirectory;
+		const publishOptions = Object.assign({
+			log: grunt.log,
+			subDirectory: ''
+		}, options.publishOptions || {});
+		options.out = grunt.option<string>('doc-dir') || options.out || grunt.config.get<string>('apiDocDirectory');
 
 		// Use project-local typedoc
 		const typedoc = require.resolve('typedoc/bin/typedoc');
+		grunt.log.writeln(`Building API Docs to "${ options.out }"`);
 		exec(`node "${ typedoc }" ${ typedocOptions(options).join(' ') }`);
 
-		// Add a .nojekyll file to prevent GitHub pages from trying to parse files starting with an underscore
-		// @see https://github.com/blog/572-bypassing-jekyll-on-github-pages
-		grunt.log.writeln(`writing .nojekyll file to ${ rootApiDocDirectory }`);
-		touch(join(rootApiDocDirectory, '.nojekyll'));
-
-		if (shouldPublish) {
+		// Publish
+		const publishMode = (typeof publishOptions.publishMode === 'function') ? publishOptions.publishMode() :
```

Comment:
This publish check is more robust than the previous `shouldPublish` check allowing better control over when we clone, commit, and publish